sign/pending_requests.go

```diff
@@ -0,0 +1,186 @@
+package sign
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/geth/account"
+)
+
+type verifyFunc func(string) (*account.SelectedExtKey, error)
+
+// PendingRequests is a capped container that holds pending signing requests.
+type PendingRequests struct {
+	mu       sync.RWMutex // to guard transactions map
+	requests map[string]*Request
+
+	log log.Logger
+}
+
+// NewRequests creates a new requests list
+func NewRequests() *PendingRequests {
+	logger := log.New("package", "status-go/sign.PendingRequests")
+
+	logger.Info("initializing transaction queue")
+	return &PendingRequests{
+		requests: make(map[string]*Request),
+		log:      logger,
+	}
+}
+
+// Reset is to be used in tests only, as it simply creates new transaction map, w/o any cleanup of the previous one
+func (rs *PendingRequests) Reset() {
+	rs.mu.Lock()
+	defer rs.mu.Unlock()
+
+	rs.requests = make(map[string]*Request)
+}
+
+// Add a new signing request.
+func (rs *PendingRequests) Add(ctx context.Context, meta Meta, completeFunc completeFunc) (*Request, error) {
+	rs.mu.Lock()
+	defer rs.mu.Unlock()
+
+	request := newRequest(ctx, meta, completeFunc)
+	rs.requests[request.ID] = request
+	rs.log.Info("signing request is created", "ID", request.ID)
+
+	go NotifyOnEnqueue(request.ID, meta, ctx)
```

Comment:
i think we are trying to move notifications as close to backend as possible

Reply:
well, it is as close to the backend as it was before.
I want to extract these signals to a separate entity and use it as a dependency, but that PR grew too big anyway.

Reply:
Essentially, you are subscribing to the requests collection events, if a sign request was created or errored, that what the API consumer (aka status-react) should know for now. It's nothing to do specifically for transactions, it just so happens that it is used to be that way. But we are really interested in double-checking the user's password on some occasions. It might be signing a transaction, it might be not :)

And that is as close (PendingRequests) is a member of a Backend anyway.

Spawning a goroutine here is a cheap an simple way to guarantee that any notification handler won't deadlock this code by, for instance, waiting for the request to be fulfilled inside.